### PR TITLE
feat: add search filter component to landing page and style it

### PR DESCRIPTION
**Pull request by ifranch**
# Pull Request Template

## Description

- Add the missing search filtering component in the hero of the landing page.
- Generate the search URL based on the filters entered.
- Update the hero so the title is in the outer section of it and can be full width, above the image.

Known issues I will try to address today or next week:
- Once button is selected, background color should be doorway-color-blue-500.
- Max monthly rent search box looks weird. Will merge the style from https://github.com/metrotranscom/doorway/pull/194
- For narrow desktop screens, the hero width looks weird and some components overlap.
- Bedroom buttons should be in one row rather than more than one. Right now there is no way to style buttons within a button group so will work on changing that.

If you see any other issues please point them out.

Desktop: 
image

Mobile: 
image
image

## How Can This Be Tested/Reviewed?

- Bring up the public site.
- In the landing page, verify that you see the search filtering component. Compare it against the mocks.
- Click on "view county map" at the bottom. This should open a modal with the county map image. Close the modal.
- Check/enter as many filters as you want and click on "View listings".
- Verify that search filtering worked as intended. You can also check that the URL reflects the filters you entered.
- Go back to the landing page and resize the screen to see how the component is updated.
- Switch to mobile view and compare the component against the mocks.

### CODE REVIEW

**Comment by phattya** [comment]
@ifranch great job!!!  I spent some time testing & functionality is working great for me!!! So excited to see this finished. Looking forward to testing it in it's entirety. Thank you!!!

File: sites/public/src/components/listings/search/ListingsSearchCombined.tsx [hunk 1]
```diff
@@ -230,6 +246,18 @@ const locations: FormOption[] = [
     label: "Sonoma",
     value: "Sonoma",
   },
+  {
+    label: "San Francisco",
+    value: "San Francisco",
+    isDisabled: true,
+    labelNoteHTML: `(For San Francisco listings, please go to <a href="https://housing.sfgov.org/">DAHLIA</a>)`,
```
Comment (phattya) [inline sites/public/src/components/listings/search/ListingsSearchCombined.tsx:253]: can we make this open in a different tab/window (ie target=_blank)?

**Comment by phattya** [comment]
@ifranch a few things I saw that can be patched up next week...
the bedroom buttons get cut off in the enabled state  (but they are so beautiful other than that!!!)
image

**Comment by isaac-looker** [comment]
clicking translate seems to crash on the modal `props.counties` being undefined. I checked #194  and it doesn't crash. (I still need to add translations on #194 ). 

counties-error-on-translate

**Comment by phattya** [comment]
And when selecting/unselecting the checkboxes the text moves back & forth...
image

**Review by phattya — APPROVED** [review]
Lgtm! We have a few fixes to do next week, but overall looking awesome!!!  I also tested the functionality over & over and it worked perfect every time.  Thanks for all your hard work on this!!!

**Comment by isaac-looker** [comment]
We'll want to remove the `search=` URI param on the redirect or fix it. 

http://localhost:3000/listings?bedrooms:1;monthlyRent:200;counties:Alameda works but 

http://localhost:3000/listings?searc=bedrooms:1;monthlyRent:200;counties:Alameda crashes with:

search-param-present

**Review by chriscasto — APPROVED** [review]
+1 to what Phatty said, and adding these:

Rent input halo gets cut off by image
image

Submit button halo gets cut off by border
image

File: sites/public/src/components/listings/search/ListingsSearchCombined.tsx (hunk 1 above)
Comment (phattya) [reply sites/public/src/components/listings/search/ListingsSearchCombined.tsx:253]: Resolving now so I can merge this in, I hope you don't mind. We can do this part next week. Thanks!!!

**Pull request merged**